### DISCUSSION TITLE
[ibus] Default config (layout: "default") seems to ignore layout_variant in certain cases, breaking non-mozc layout.

### DISCUSSION THREAD

**Discussion by ghost** [google/mozc#1142] [Q&A] [answered]
## Description
I have a problem that requires the following example workaround in ibus_config.textproto on Ubuntu 24.04 with Gnome (Wayland):
```
engines {
  name : "mozc-jp"
  longname : "Mozc"
  layout : "us"
  layout_variant: "altgr-intl"
}
```

## Steps to reproduce
1. Have us+altgr-intl as a first layout (I believe it's not specific to altgr-intl variant), mozc-jp as second layout
2. Switch to mozc-jp normally
3. Use some tool like ydotool to generate a keypress from virtual keyboard, that triggers layout switch shortcut in Gnome. May require changing custom shortcuts to reproduce.

## Expected behavior
Should switch to us+altgr-intl like expected (or any other layout that is set up). It does normally happen, but under some cirtumstances, it does not.

## Actual behavior
Layout becomes simple US layout, without any AltGr layers. More importantly, it doesn't go back, no matter what I do. Only logout helps.

## Screenshots
Not necessary, I believe.

## Version or commit-id
Mozc-2.28.4715.102+24.11.oss

## Environment
 - OS: Ubuntu 24.04
 - IMF (for Linux): Ibus
 - Related Applications: Any

## Investigations
 - Whether this issue happens on both Ibus and other IMF (e.g. Fcitx, uim).
   + N/A
 - Whether this issue happens on other IMEs (e.g. Anthy, SKK).
   + N/A
 - What applications this issue happens on (e.g. Chromium, gedit).
   + seemingly any
 - What applications this issue does not happen on (e.g. Chromium, gedit).
   + not-found
 - (optional) What versions or commit-ids this issue did not happen on
   (e.g. Mozc-2.28.4960.100+24.11.oss).
   + no-investigation.

**Comment by yukawa** [comment]
> I have a problem that requires the following example workaround in ibus_config.textproto on Ubuntu 24.04 with Gnome (Wayland):
>
> ```engines {
>  name : "mozc-jp"
>  longname : "Mozc"
>  layout : "us"
>  layout_variant: "altgr-intl"
>}
>```

Hmm, was there any way to solve your issue without hard-coding `us` and  `altgr-intl`?  If that's the only way that is currently available, it sounds like there is nothing we can do in the Mozc repository. Can you check if the same behavior is observed or not with other ibus-based IMEs? If other IMEs behave in the same way, I believe this would be one of the use cases that needs to be recognized and officially supported by the Gnome (and ibus) community.

Let us know if you find a way to solve your issue with reasonable default values.

**Reply by ghost** [reply]
I've reverted it to default for now, I'll watch. Maybe it was my fault for triggering some other bug that happens to mess with key mappings, I'm investigating. I managed to trigger that bug without mozc-jp added as a layout at all.

**Reply by ghost** [reply, marked as answer]
Yeah, it seems like I had a messy system settings after all. Old xkb tweaks which were non-functional after all. Thank you for patience and attention to the issue.
